pkg/fanal/analyzer/language/elixir/mixlock/mixlock_test.go

```diff
@@ -0,0 +1,89 @@
+package mixlock
+
+import (
+	"github.com/aquasecurity/trivy/pkg/fanal/analyzer"
+	"github.com/aquasecurity/trivy/pkg/fanal/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"os"
+	"testing"
+)
+
+func Test_elexirLockAnalyzer_Analyze(t *testing.T) {
+	tests := []struct {
+		name      string
+		inputFile string
+		want      *analyzer.AnalysisResult
+	}{
+		{
+			name:      "happy path",
+			inputFile: "testdata/happy.mix.lock",
+			want: &analyzer.AnalysisResult{
+				Applications: []types.Application{
+					{
+						Type:     types.Hex,
+						FilePath: "testdata/happy.mix.lock",
+						Libraries: []types.Package{
+							{
+								ID:        "bunt@0.2.0",
+								Name:      "bunt",
+								Version:   "0.2.0",
+								Locations: []types.Location{{StartLine: 2, EndLine: 2}},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name:      "empty file",
+			inputFile: "testdata/empty.mix.lock",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			f, err := os.Open(tt.inputFile)
+			require.NoError(t, err)
+			defer func() {
+				err = f.Close()
+				assert.NoError(t, err)
+			}()
+
+			a := mixLockAnalyzer{}
+			got, err := a.Analyze(nil, analyzer.AnalysisInput{
+				FilePath: tt.inputFile,
+				Content:  f,
+			})
+
+			assert.NoError(t, err)
+			assert.Equal(t, tt.want, got)
+		})
+	}
+}
+
+func Test_elexirLockAnalyzer_Required(t *testing.T) {
+	tests := []struct {
+		name     string
+		filePath string
+		want     bool
+	}{
+		{
+			name:     "happy path",
+			filePath: "mix.lock",
+			want:     true,
+		},
```

Comment:
If Mix accepts `*mix.lock`, we should add such a test as well.

Reply:
looks like lock file doesn't have rules for name. Name may have any name.
https://hexdocs.pm/mix/Mix.Project.html#module-configuration.

I added footnote on docs for mix(as for conan).
https://github.com/aquasecurity/trivy/blob/afd32a36284e8a90371ff6cf429767ee6a68ef49/docs/docs/vulnerability/detection/language.md?plain=1#L30

I think this is better way.

Reply:
We usually support the default name only. If people want to use other names, they can specify `--file-pattern`. It makes sense to support only `mix.lock`.

>:lockfile - the name of the lockfile used by the mix deps.* family of tasks. Defaults to "mix.lock".